### ISSUE TITLE
SC2004 detected with FOR loop

### ISSUE THREAD

**Issue by neyaki** [koalaman/shellcheck#2694]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version: "online" 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
# set counter 'c' to 1 and condition 
# c is less than or equal to 5
i=5
for (( c=1; c<=$i; c++ ))
do 
   echo "Welcome $c times"
done

```

#### Here's what shellcheck currently says:
[Line 5:](javascript:setPosition(5, 16))
for (( c=1; c<=$i; c++ ))
               ^-- [SC2004](https://www.shellcheck.net/wiki/SC2004) (style): $/${} is unnecessary on arithmetic variables.

#### Here's what I wanted or expected to see:
Nothing, as it's an acceptable way to perform a FOR loop

**Comment by thedward** [comment]
The provided link for [SC2004](https://www.shellcheck.net/wiki/SC2004) explains the rationale for the rule. If you'd prefer not to get that warning, you can disable that rule with a shellcheck [directive ](https://www.shellcheck.net/wiki/Directive):

```
#!/bin/bash
# set counter 'c' to 1 and condition 
# c is less than or equal to 5
i=5
# shellcheck disable=SC2004
for (( c=1; c<=$i; c++ ))
do 
   echo "Welcome $c times"
done
```

**Comment by port19x** [comment]
You might also want to run with a different severity. 
e.g. `shellcheck -S info` since your comment of $i being an "acceptable way" makes me assume you wouldn't like some of the other style warnings either
